src/app/dynamic/enterprise/quotas/project-quota-dialog/component.ts

```diff
@@ -0,0 +1,154 @@
+import {ProjectService} from '@core/services/project';
+import {QuotaService} from '@core/services/quota';
+import {QuotaVariables, QuotaDetails, Quota} from '@shared/entity/quota';
+import {KmValidators} from '@shared/validators/validators';
+import {ControlsOf} from '@shared/model/shared';
+import {Project} from '@shared/entity/project';
+
+enum Error {
+  Required = 'required',
+  AtLeastOneRequired = 'atLeastOneRequired',
+}
+
+@Component({
+  selector: 'km-admin-dialog',
+  templateUrl: './template.html',
+  styleUrls: ['./style.scss'],
+})
+export class ProjectQuotaDialogComponent implements OnInit, OnDestroy {
+  private _unsubscribe = new Subject<void>();
+
+  readonly Error = Error;
+
+  form: FormGroup<ControlsOf<Quota>>;
+
+  projects: Project[] = [];
+  quotas: QuotaDetails[] = [];
+  selectedProject: Project;
+  selectedQuota: QuotaDetails;
+
+  constructor(
+    private readonly _dialogRef: MatDialogRef<ProjectQuotaDialogComponent>,
+    private readonly _notificationService: NotificationService,
+    private readonly _projectService: ProjectService,
+    private readonly _quotaService: QuotaService,
+    private readonly _builder: FormBuilder
+  ) {}
+
+  get quotaGroup(): FormGroup<ControlsOf<QuotaVariables>> {
+    return this.form?.controls.quota;
+  }
+
+  ngOnInit(): void {
+    this._initForm();
+    this._getProjects();
+    this._getQuotas();
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  resetSubjectNameControl() {
+    this.form.controls.subjectName.reset();
+  }
+
+  getObservable(): Observable<Record<string, never>> {
+    if (this.form.invalid) return of(null);
+
+    const formValue = this.form.value as Quota;
+
+    const quota: QuotaVariables = Object.fromEntries(Object.entries(formValue.quota).filter(([_, v]) => !!v));
+
+    const update$ = this._quotaService.updateQuota(this.selectedQuota?.name, quota);
+
+    const create$ = this._quotaService.createQuota({...formValue, quota});
+
+    return (this.selectedQuota ? update$ : create$).pipe(
+      tap({
+        next: () => {
+          this._dialogRef.close(true);
+          this._notificationService.success(
+            `${this.selectedQuota ? 'Updated' : 'Created'} project quota for ${formValue.subjectName}`
+          );
+          this._quotaService.refreshQuotas();
+        },
+      })
+    );
+  }
+
+  private _getProjects(): void {
+    this._projectService.allProjects.pipe(takeUntil(this._unsubscribe)).subscribe(projects => {
+      this.projects = projects;
+      this.form.controls.subjectName.enable();
+    });
+  }
+
+  private _getQuotas(): void {
+    this._quotaService.quotas.pipe(takeUntil(this._unsubscribe)).subscribe(quotas => {
+      this.quotas = quotas;
+    });
+  }
+
+  private _initForm(): void {
+    this.form = this._builder.group<ControlsOf<Quota>>({
+      quota: this._builder.group<ControlsOf<QuotaVariables>>(
+        {
+          cpu: this._builder.control(''),
```

Comment:
@lsviben @pkprzekwas @Waseem826  any thoughts?

- secondly, IMO, we should add default `unit` on Label so that user knows what is the default unit.

**e.g** 
- RAM -> 100Mi
- DISK -> 50G ?

Reply:
I am working on making the RAM and Disk to be set to GB for the API . So the value will be a float in the end and UI doesnt have to worry about the units, they will just get float numbers